core/src/test/java/org/apache/calcite/avatica/AvaticaResultSetThrowsSqlExceptionTest.java

```diff
@@ -75,11 +90,118 @@ public void testUpdateNull() throws SQLException {
     final TestDriver driver = new TestDriver();
     try (Connection connection = driver.connect("jdbc:test", properties);
          ResultSet resultSet =
-             connection.createStatement().executeQuery("SELECT * FROM TABLE")) {
+                 connection.createStatement().executeQuery("SELECT * FROM TABLE")) {
       thrown.expect(SQLFeatureNotSupportedException.class);
       resultSet.updateNull(1);
     }
   }
-}
 
+  @Test
+  public void testCommonCursorStates() throws SQLException {
+    final ResultSet resultSet = getResultSet();
+
+    // right after statement execution, result set is before first row
+    assert resultSet.isBeforeFirst();
+
+    // retrieve each row until the last one
+    while (!resultSet.isAfterLast()) {
+      assert resultSet.next() != resultSet.isAfterLast();
```

Comment:
Please avoid using Java asserts in the test code. Test failures will be hard to understand/analyze in case of assert failures.

Reply:
I have replaced them with the JUnit counterparts (e.g., Assert.assertTrue), this would do?